torch/csrc/jit/script/schema_matching.cpp

```diff
@@ -299,19 +299,31 @@ c10::optional<MatchedSchema> tryMatchSchema(
       return c10::nullopt;
     }
   }
-  auto return_types = fmap(schema.returns(), [&](const Argument& r) {
+
+  const auto &returns = schema.returns();
+  auto return_types = fmap(returns, [&](const Argument& r) {
     return evalTypeVariables(r.type(), type_env);
   });
-  return MatchedSchema{std::move(positional_inputs), std::move(return_types)};
+  // Codegen does not support return of namedtuples with undefined field names.
+  // Therefore, either all or none returns has field names.
+  bool return_has_field_names = std::all_of(returns.begin(), returns.end(),
+    [&](const Argument& r) { return r.name().length() > 0; });
+  c10::OptNameList return_field_names = c10::nullopt;
+  if (return_has_field_names) {
+    return_field_names = fmap(returns, [&](const Argument& r) {
+      return r.name();
+    });
+  }
+  return MatchedSchema{std::move(positional_inputs), std::move(return_types), std::move(return_field_names)};
```

Comment:
Wait, what? So script functions now return named tuples too?

Reply:
No, script functions does not return named tuples, but the name information is kept inside `TupleType`, and accessible only inside JIT when inlined. See the other PR:
https://github.com/pytorch/pytorch/pull/16505/files#diff-78aab5b4217aeaed920efab0c5f5f5c8R8282